### PR TITLE
Fix Guava import version

### PR DESCRIPTION
**Pull request by chanseokoh**

### CODE REVIEW

**Comment by chanseokoh** [comment]
I remember @briandealwis added some kind of test a while ago (https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/pom.xml#L447), but I guess this isn't really working.

**Comment by briandealwis** [comment]
It is frustrating.  The rule is succeeding as it tests the version that is resolved rather than the potential versions that can be resolved.  We're resolving to Guava 20.0, which satisfies both the package constraint (`version="15.0.0"`) and passes the dependency rule.

**Comment by elharo** [comment]
Then let's add a more straightforward test that simply reads the files directly rather than trying to resolve the dependency.

**Comment by chanseokoh** [comment]
The test is not flexible, but it should serve its purpose. One caveat is that not every bundle has `PluginXmlTest`. (Some bundles don't have `plugin.xml`.)

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java [hunk 1]
```diff
@@ -189,6 +192,35 @@ public final void testBundleVendor() throws IOException {
...
+    Pattern pattern = Pattern.compile(regexPrefix + "[^;,]+");
+
+    Matcher matcher = pattern.matcher(value);
+    while (matcher.find()) {
+      int nextCharOffset = matcher.end();
+      if (nextCharOffset == value.length()) {
+        fail("Import-Package version not defined");
+      }
+      String stringAfterMatch = value.substring(nextCharOffset);
+      if (!stringAfterMatch.startsWith(";version=" + versionString)) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: Hmm, this doesn't seem to match the callers above?

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java [hunk 2]
```diff
@@ -189,6 +192,35 @@ public final void testBundleVendor() throws IOException {
...
+  private void checkVersionStringsInManifestMf(
+      String value, String prefixToCheck, String versionString) {
+    String regexPrefix = prefixToCheck.replaceAll(".", "\\\\.");
+    Pattern pattern = Pattern.compile(regexPrefix + "[^;,]+");
+
+    Matcher matcher = pattern.matcher(value);
+    while (matcher.find()) {
+      int nextCharOffset = matcher.end();
+      if (nextCharOffset == value.length()) {
+        fail("Import-Package version not defined");
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: This is called above for Require-Bundles too.

**Comment by chanseokoh** [comment]
Oops, the errors due to pushing too prematurely to head home.

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java [hunk 3]
```diff
@@ -189,6 +192,35 @@ public final void testBundleVendor() throws IOException {
     assertEquals("Google Inc.", vendor);
   }
 
+  @Test
+  public final void testGuavaImportVersions() throws IOException {
+    checkVersionStringsInManifestMf(
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: This looks really useful.  Could I suggest a refactoring?
  - call this `testDependencyVersions()`
  - add methods `checkImportPackageVersion(package-pattern, version-range)` and `checkRequireBundleVersion(bundle-name, version-range)` (e.g., `checkImportPackageVersion("com.google.common.*", "[20,21)")`) — embedding `"Require-Bundle"` or `"Import-Package"` and `"version"` vs `"bundle-version"` is error prone
  - use `org.osgi.framework.VersionRange#valueOf(String)` to parse and compare the version ranges
  - the patterns should allow `\s` for white-space

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java [hunk 4]
```diff
@@ -189,6 +192,35 @@ public final void testBundleVendor() throws IOException {
...
+  }
+
+  private void checkVersionStringsInManifestMf(
+      String attributeName, String prefixToCheck, String versionString) throws IOException {
+    String value = getManifestAttributes().getValue(attributeName);
+    if (value != null) {
+      String regexPrefix = prefixToCheck.replaceAll("\\.", "\\\\.");
+      Pattern pattern = Pattern.compile(regexPrefix + "[^;,]*");
+
+      Matcher matcher = pattern.matcher(value);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: Might be easier to use `Splitter.on(',')` and then apply the pattern to each import directive?

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: I would have gone that way if it worked. Commas can be in, e.g., `version="[20.0.0,21.0.0)"`.

Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: Ah good point :-)

File: plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java (hunk 3 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/BasePluginXmlTest.java]: Good idea. But I am not thrilled to spend more time on this to make it general and flexible ATM. If there is unnecessary white-space, let's just remove it and be consistent.

**Pull request merged**